rectools/model_selection/random_split.py

```diff
@@ -0,0 +1,127 @@
+    filter_cold_users: bool, default ``True``
+        If `True`, users that not in train will be excluded from test.
+    filter_cold_items: bool, default ``True``
+        If `True`, items that not in train will be excluded from test.
+    filter_already_seen: bool, default ``True``
+        If `True`, pairs (user, item) that are in train will be excluded from test.
+
+    Examples
+    --------
+    >>> from rectools import Columns
+    >>> df = pd.DataFrame(
+    ...     [
+    ...         [1, 2, 1, "2021-09-01"],  # 0
+    ...         [2, 1, 1, "2021-09-02"],  # 1
+    ...         [2, 3, 1, "2021-09-03"],  # 2
+    ...         [3, 2, 1, "2021-09-03"],  # 3
+    ...         [3, 3, 1, "2021-09-04"],  # 4
+    ...         [3, 4, 1, "2021-09-04"],  # 5
+    ...         [1, 2, 1, "2021-09-05"],  # 6
+    ...         [4, 2, 1, "2021-09-05"],  # 7
+    ...     ],
+    ...     columns=[Columns.User, Columns.Item, Columns.Weight, Columns.Datetime],
+    ... ).astype({Columns.Datetime: "datetime64[ns]"})
+    >>> interactions = Interactions(df)
+    >>>
+    >>> rs = RandomSplitter(test_size=0.25, random_state=42, n_splits=2, filter_cold_users=False,
+    ...                     filter_cold_items=False, filter_already_seen=False)
+    >>> for train_ids, test_ids, _ in rs.split(interactions):
+    ...     print(train_ids, test_ids)
+    [1 2 3 4 5 7] [0 6]
+    [0 1 2 4 5 6] [3 7]
+    >>>
+    >>> rs = RandomSplitter(test_size=0.25, random_state=42, n_splits=2, filter_cold_users=True,
+    ...                     filter_cold_items=True, filter_already_seen=True)
+    >>> for train_ids, test_ids, _ in rs.split(interactions):
+    ...     print(train_ids, test_ids)
+    [1 2 3 4 5 7] []
+    [0 1 2 4 5 6] [3]
+    """
+
+    def __init__(
+        self,
+        test_size: float,
+        n_splits: int = 1,
+        random_state: Optional[int] = None,
+        filter_cold_users: bool = True,
+        filter_cold_items: bool = True,
+        filter_already_seen: bool = True,
+    ) -> None:
+        if test_size <= 0.0 or test_size >= 1.0:
+            raise ValueError("Value of test_size must be between 0 and 1")
+
+        super().__init__()
+        self.random_state = random_state
+        self.n_splits = n_splits
+        self.test_size = test_size
+        self.filter_cold_users = filter_cold_users
+        self.filter_cold_items = filter_cold_items
+        self.filter_already_seen = filter_already_seen
+
+    def _split_without_filter(
+        self,
+        interactions: Interactions,
+        collect_fold_stats: bool = False,
+    ) -> tp.Iterator[tp.Tuple[np.ndarray, np.ndarray, tp.Dict[str, tp.Any]]]:
+        rng = np.random.default_rng(self.random_state)
+        df = interactions.df
+        idx = pd.RangeIndex(0, len(df))
+
+        test_part_size = int(round(self.test_size * len(df)))
+        if test_part_size == 0:
+            raise ValueError("Test part must be not empty")
```

Comment:
Let's give more informative error message. `test_part` is given float parameter. If user set it as `0.001` and we calculated here `test_part_size = 0`, user will not understand the meaning of the message

And the next message too

Reply:
Done